#! /usr/bin/env python

PACKAGE='tuw_geometry'
from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

gen.add("line_dection_split_neighbor", bool_t, 0, " ", False)
gen.add("line_dection_split_threshold", double_t, 0, "Split threshold", .05, 0, 0.5)
gen.add("line_dection_min_length", double_t, 0, "Min line length", 0.5, 0, 5)
gen.add("line_dection_min_points_per_line", int_t, 0, "Minimum points on line", 40, 0, 100)
gen.add("line_dection_min_points_per_unit", double_t, 0, "Minimum number of points per unit", 10, 0, 100)

exit(gen.generate(PACKAGE, "tuw_geometry", "Linesegment2DDetector"))